### PR TITLE
Add test 'Filter, sort and pagination search engines'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add test 'Filter, sort and pagination search engines'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `URL_FO=shopUrl/ TEST_PATH='functional/BO/13_shopParameters/06_trafficAndSeo/02_searchEngines/01_filterSortAndPagination.js' npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21405)

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/13_shopParameters/06_trafficAndSeo/02_searchEngines/01_filterSortAndPagination.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+const seoAndUrlsPage = require('@pages/BO/shopParameters/trafficAndSeo/seoAndUrls');
+const searchEnginesPage = require('@pages/BO/shopParameters/trafficAndSeo/searchEngines');
+
+// Import data
+const {searchEngines} = require('@data/demo/searchEngines');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_shopParameters_TrafficAndSeo_searchEngines_filterSortAndPagination';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/13_shopParameters/06_trafficAndSeo/02_searchEngines/01_filterSortAndPagination.js]:
```suggestion
const baseContext = 'functional_BO_shopParameters_trafficAndSeo_searchEngines_filterSortAndPagination';
```

**Review by Progi1984 — COMMENTED** [review]
A small comment

**Review by Progi1984 — DISMISSED** [review]

File: tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/index.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  /**
+   * Select pagination limit
+   * @param page
+   * @param number
+   * @returns {Promise<string>}
+   */
+  async selectPaginationLimit(page, number) {
+    await page.click(this.paginationDropdownButton);
+    await this.clickAndWaitForNavigation(page, this.paginationItems(number));
+    return this.getPaginationLabel(page);
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(page, this.paginationItems(number));
    
    return this.getPaginationLabel(page);
```

File: tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/index.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  /**
+   * Click on next
+   * @param page
+   * @returns {Promise<string>}
+   */
+  async paginationNext(page) {
+    await this.clickAndWaitForNavigation(page, this.paginationNextLink);
+    return this.getPaginationLabel(page);
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(page, this.paginationNextLink);
    
    return this.getPaginationLabel(page);
```

File: tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/index.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  /**
+   * Click on previous
+   * @param page
+   * @returns {Promise<string>}
+   */
+  async paginationPrevious(page) {
+    await this.clickAndWaitForNavigation(page, this.paginationPreviousLink);
+    return this.getPaginationLabel(page);
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/shopParameters/trafficAndSeo/searchEngines/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(page, this.paginationPreviousLink);
    
    return this.getPaginationLabel(page);
```

**Review by NeOMakinG — COMMENTED** [review]
Some comments to improve readability

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**